app/services/entryServices.js

```diff
@@ -0,0 +1,58 @@
+import Entry from '../models/Entry';
+import { AppError } from '@/utils/errorHandler';
+import { dbConnect } from '@/utils/dbConnects';
+
+export async function getEntriesByCompetition(competitionId) {
+  await dbConnect();
+  return await Entry.find({ competition: competitionId }).sort({
+    createdAt: -1,
+  });
+}
+export async function createEntry({
+  competition,
+  participantId,
+  imageUrl,
+  caption,
+}) {
+  await dbConnect();
+  if (!competition) {
+    throw new AppError('Competition is required.', 400);
```

Comment:
Please don't throw exceptions. This will cause the server to crash. Return an error in the response instead